src/config.rs

```diff
@@ -65,8 +66,18 @@ impl Environment {
         self.get_values_for("BUILD_STD", target, bool_from_envvar)
     }
 
-    fn image(&self, target: &Target) -> Option<String> {
+    fn image(&self, target: &Target) -> Result<Option<PossibleImage>> {
```

Comment:
If this can't fail, why change this to a `Result`? It avoids having `let env = self.env.image(target)?;` down below, and can just keep the old code.

The code can still be similar, as:

```rust
    fn image(&self, target: &Target) -> Option<PossibleImage> {
        self.get_target_var(target, "IMAGE")
            .map(|i| i.parse().expect("infallible"))
            .map(|mut i: PossibleImage| {
                if let Some(toolchain) = self.get_target_var(target, "IMAGE_TOOLCHAIN") {
                    i.toolchain = Some(ImageArchitecture::from_target(toolchain.into()));
                    i
                } else {
                    i
                }
            })
    }

...

    pub fn image(&self, target: &Target) -> Result<Option<PossibleImage>> {
        self.get_config(target, Environment::image, CrossToml::image)
    }
```

Reply:
yep, I changed it by mistake

Reply:
this can fail now